src/private/wait.rs

```diff
@@ -0,0 +1,97 @@
+use core::time::Duration;
+#[cfg(feature = "std")]
+use std::sync::{Condvar, Mutex};
+
+use embedded_svc::mutex::Mutex as _;
+#[cfg(not(feature = "std"))]
+use esp_idf_sys::*;
+
+#[cfg(not(feature = "std"))]
+use super::time::micros_since_boot;
+
+pub struct Waiter {
+    #[cfg(feature = "std")]
+    cvar: Condvar,
+    #[cfg(feature = "std")]
+    running: Mutex<bool>,
```

Comment:
If you are still willing to drive e.g. the `sntp` implementation which needs (?) a "waiter", here's one that (ideally) should work for sntp & ping, but also for esp, wifi:
https://github.com/esp-rs/esp-idf-svc/compare/waitable

Not really tested but at least compiles and passed fmt and clippy checks. And used in all of ping, eth and wifi. Hopefully that implementation makes clearer some of the changes I'm asking for here.

Reply:
> If you are still willing to drive e.g. the sntp implementation which needs (?) a "waiter", here's one that (ideally) should work for sntp & ping, but also for eth & wifi:

Oh, you wrote your own now?
Should I close this then?

Reply:
Up to you. You can also take bits and pieces of mine and update yours with these. The reason I took the effort to actually write something is because you might feel that I'm nitpicking on everything which is definitely not the impression I want to leave - so I thought - I can at least take the effort to justify -in code - including how it will be used in eth and wifi code - why the waiter needs to be generified, you know. :)